lib/OpenQA/BuildResults.pm

```diff
@@ -85,13 +85,29 @@ sub add_review_badge {
 }
 
 sub compute_build_results {
-    my ($group, $limit, $time_limit_days, $tags) = @_;
+    my ($group, $limit, $time_limit_days, $tags, $subgroups) = @_;
 
+    # find child groups
     my $group_ids;
     my @children;
     if ($group->can('children')) {
-        @children  = $group->children;
-        $group_ids = $group->child_group_ids;
+        if (@$subgroups) {
+            # filter subgroups
+            $group_ids = [];
+            my $group_name = $group->name;
+            for my $child ($group->children) {
+                my $full_name = $child->full_name;
+                if (grep { $_ eq '' || $full_name =~ /$_/ } @$subgroups) {
+                    push(@$group_ids, $child->id);
+                    push(@children,   $child);
+                }
+            }
+
+        }
+        else {
+            $group_ids = $group->child_group_ids;
+            @children  = $group->children;
+        }
```

Comment:
IMHO getting too complicated. I would at least extract a method. You have a name for it already: "filter_subgroups" ;)